### DISCUSSION TITLE
Consuming a previously synthesised CDK Cloud Assembly and define environment config at deploy?

### DISCUSSION THREAD

**Discussion by vpste1** [aws/aws-cdk#16082] [Q&A]
### Goal

The end goal I'm trying to achieve is: with a previously synthesised (`cdk synth`) CDK Cloud Assembly (`cdk.out`) I want to go and deploy it to several stages, each with their own environment config. 

I'm familiar with how CDK pipelines can do this with an all-in-one by binding an application stack's construct to a stage via `pipeline.addStage`:

```jsx
import { ApplicationStack } from "./application-stack";

export class PipelineStage extends Stage {
...
    // Application stack bound to stage here:
    const service = new ApplicationStack(this, "WebService");
  }
}
```

```jsx
import { PipelineStage } from './pipeline-stage.ts'

export class PipelineStack extends Stack {
...
    const pipeline = new CodePipeline(this, 'Pipeline', {
			...
		),
  });

	pipeline.addStage(new PipelineStage(this, 'PreProd', {
	  env: { account: 'ACCOUNT1', region: 'us-east-2' }
	}));
...
}
```

But this doesn't allow for consuming a CDK Cloud Assembly artefact.

### Previous attempts/ideas

Use a Stage construct:
I was hoping that I could read a Cloud Assembly from a local directory as PoC, but don't know how to derive a `cloudAssemblyArtifact` from our imported `webstaticAssembly` below (or if it's even possible?):
```jsx
const webStaticAssembly = new CloudAssembly(
  path.resolve(__dirname, STATIC_WEB_ASSEMBLY_PATH)
);

new CdkStage(pipeline, "PreProd", {
  // * The CodePipeline Artifact with the Cloud Assembly.
  cloudAssemblyArtifact: CA_ARTIFACT,
});
```

Consume the Cloud Formation template:
The Cloud Assembly contains the Cloud Formation template (amongst other assets), I was thinking I could potentially define a Stack that is derived from an applications CF template; this however felt wrong, and asset management was getting tricky. I must admit I stopped exploring this path early as I felt I was bending a Cloud Assembly to something it wasn't built for (and defeating the purpose of building into a Cloud Assembly in the first place).

Straight from CLI whilst utilising Cloud Formation Parameters:
I'm familiar with deploying an already synthesised application from the CLI like so:

`cdk --app ./my-cloud-assembly ls`

And hoped it could be combined with specified parameters (e.g an stage name); however unresolved tokens in a Construct's ID isn't allowed (and I didn't find any escape hatches to get around it):

`Cannot use tokens in construct ID: ${Token[TOKEN.238]}`

### The question

Is what I am trying to do possible? Have I overlooked something entirely?

**Comment by peterwoodworth** [comment]
I don't think this is possible unfortunately. The cloud assembly is already synthesized and ready to go, implementing a way to adjust some aspects of it does sound against its design to me.
